lib/android-helpers.js

```diff
@@ -89,43 +89,7 @@ helpers.ensureNetworkSpeed = function (adb, networkSpeed) {
 };
 
 helpers.ensureDeviceLocale = async function (adb, language, country) {
-  let haveLanguage = language && typeof language === "string";
-  let haveCountry = country && typeof country === "string";
-  if (!haveLanguage && !haveCountry) {
-    return;
-  }
-  let changed = false;
-  if (await adb.getApiLevel() < 23) {
-    let curLanguage = await adb.getDeviceLanguage();
-    let curCountry = await adb.getDeviceCountry();
-    if (haveLanguage && language !== curLanguage) {
-      await adb.setDeviceLanguage(language);
-      changed = true;
-    }
-    if (haveCountry && country !== curCountry) {
-      await adb.setDeviceCountry(country);
-      changed = true;
-    }
-  } else { // API >= 23
-    let curLocale = await adb.getDeviceLocale();
-    let locale;
-    if (!haveCountry) {
-      locale = language.toLowerCase();
-    } else if (!haveLanguage) {
-      locale = country;
-    } else {
-      locale = `${language.toLowerCase()}-${country.toUpperCase()}`;
-    }
-
-    logger.debug(`Current locale: '${curLocale}'; requested locale: '${locale}'`);
-    if (locale !== curLocale) {
-      await adb.setDeviceLocale(locale);
-      changed = true;
-    }
-  }
-  if (changed) {
-    await adb.reboot();
-  }
+  return await adb.setDeviceLanguageCountry(language, country);
```

Comment:
Or should I implement an assertion whether setting locale succeeds or not? 🤔 

```js
await adb.setDeviceLanguageCountry(language, country);
  // haveLanguage and haveCountry
if (await adb.getApiLevel() < 23) {
  let curLanguage = await adb.getDeviceLanguage();		
  let curCountry = await adb.getDeviceCountry();
  if ((language === curLanguage) && (country === curCountry)) {
  } else {
    // throw some exceptions for failing setting locale.
  }
} else {
  if (`${language}-${country}` === curLocale)  {
    const curLocale = await adb.getDeviceLodale();
    // throw some exceptions for failing setting locale.
  }
}
return;
```

I consider a bit...